src/backend/cuda/platform.cpp

```diff
@@ -555,7 +565,7 @@ int DeviceManager::setActiveDevice(int device, int nId)
 
     int numDevices = cuDevices.size();
 
-    if (device > numDevices)
+    if (device >= numDevices)
```

Comment:
This seems like a bug that should be fixed in another PR.

Reply:
Wouldn't having it in a separate commit suffice ? I will send out another PR, however I don't get the point of it because we don't do merge commits anyway.